ui/app/templates/rule_modal.html

```diff
@@ -47,15 +47,31 @@ <h3 class="modal-title" ng-show="!is_edit && is_duplicate">Duplicate Existing Ru
       <div class="col-md-6">
         <div class="form-group" ng-class="{'has-error': errors.backgroundRate}">
           <label for="id_rate">Rate</label>
-          <input type="text" class="form-control" id="id_rate" ng-model="rule.backgroundRate">
-          <p class="help-block" ng-show="errors.backgroundRate">{{ errors.backgroundRate.join(', ') }}</p>
+          <input type="text"
+                 class="form-control"
+                 ng-class="{'validation_errors': integer_validation_errors.rate}"
+                 name="rate"
+                 id="id_rate"
+                 ng-model="rule.backgroundRate"
+                 required>
+          <div class="form-group" ng-class="{'has-error': integer_validation_errors.rate}" ng-show="integer_validation_errors.rate">
```

Comment:
What's the purpose of the `<div>`? Why not just apply the ng-class to the `<p>` directly?

Reply:
I was having trouble with the CSS Specificity in applying the error styling to the `<p>` elements directly. I have somehow managed to get rid of the `<div>` elements by adding a `p.has-error` css definition that takes care of the specificity issues. Is there a better way?